backend/coreapp/serializers.py

```diff
@@ -212,24 +215,52 @@ class Meta:
         read_only_fields = ["last_pulled", "is_pulling"]
 
 
-class ProjectSerializer(serializers.ModelSerializer[Project]):
-    slug = serializers.SlugField(read_only=True)
+class ProjectSerializer(JSONFormSerializer, serializers.ModelSerializer[Project]):
+    slug = serializers.SlugField()
     url = HyperlinkedIdentityField(view_name="project-detail")
     html_url = HtmlUrlField()
-    repo = GitHubRepoSerializer(read_only=True)
+    repo = GitHubRepoSerializer()
     members = SerializerMethodField()
+    most_common_platform = SerializerMethodField()
+    unmatched_function_count = SerializerMethodField()
 
     class Meta:
         model = Project
         exclude: List[str] = []
         depth = 1  # repo
 
+    def create(self, validated_data: Any) -> Project:
+        repo_data = validated_data.pop("repo")
+        repo = GitHubRepo.objects.create(**repo_data)
+        project = Project.objects.create(repo=repo, **validated_data)
+        return project
+
+    def update(self, instance: Project, validated_data: Any) -> Project:
+        for attr, value in validated_data.items():
+            setattr(instance, attr, value)
+        instance.save()
+        return instance
+
     def get_members(self, project: Project) -> List[Dict[str, Any]]:
         return [
-            serialize_profile(self.context["request"], member.profile, True)
+            serialize_profile(self.context["request"], member.user.profile, True)
             for member in project.members()
         ]
 
+    def get_most_common_platform(self, project: Project) -> Optional[str]:
```

Comment:
Can you explain the motivation for this? Why would a project even have more than one platform? I think if you're decompiling a game that shares platforms, those should be considered multiple projects on the site. Maybe there's a scenario you had in mind I'm not seeing though

Reply:
It's basically just to show the game's platform on the frontend, "most common" here almost certainly always means _the_ platform, it's just describing what the method actually does. Like Github does with repo languages

Reply:
okay. I think it's needlessly complex though. A project can just have one platform, imo. doing this whole calculation is unnecsesary. We can just grab the first object and use its platform

Reply:
> We can just grab the first object and use its platform

done :+1: